### ISSUE TITLE
Enquêter sur la création d'utilisateurs automatiques sur data gouv

### ISSUE THREAD

**Issue by maudetes** [etalab/data.gouv.fr#749]

**Comment by abulte** [comment]
Analyse des users du jour :
- sur 40 utilisateurs créées, 30 sont louches
- les comptes ne sont pas confirmés
- les comptes ne comportent aucune description, site web, image... qui pourrait laisser passer à du spam SEO
- les comptes sont créés à plusieurs minutes, voire dizaines de minutes d'intervalle (plus clickfarm que bot)
- les IPs utilisées sont similaires mais changeantes, toutes chez Proline, certaines russes
- les IPs sont connues de certaines bases comme envoyant du spam

Je ne vois vraiment pas l'intérêt de l'attaque. En revanche il est clair que nous envoyons du spam aux personnes inscrites.

Remédiations possibles :
- captcha (et encore, si c'est du clickfarm...)
- WAF qui détecterait les adresses IP suspectes
- ? @jordanguedj

**Comment by abulte** [comment]
💡 L'idée c'est peut-être de checker si les comptes existent sur la plateforme, puisque le cas échéant nous affichons :

> L'adresse XXX est déjà utilisée.

A partir de là, on peut penser qu'une liste de mot de passe leakée et/ou du bruteforce est utilisée pour tenter de prendre le contrôle du compte.

**Comment by abulte** [comment]
Le pattern semble actif depuis au moins novembre dernier et concerne plus de 1000 comptes.

```
>>> us = User.objects.filter(last_name__endswith='KM')
>>> for u in us:
...     print(u.slug, u.created_at)
...
duklidastooguekm-vfkfybzstooguekm 2022-03-17 10:24:34.251000
uriserstooguekm-vladimirstooguekm 2022-03-17 10:03:31.543000
...
adoratiusstooguekm-kostyastooguekm 2021-10-31 11:32:59.948000
rfnczstooguekm-yfcntyfstooguekm 2021-10-31 11:32:24.026000
```

**Comment by maudetes** [comment]
Pour info, il y a d'autres patterns existants, ex [XB](https://object.files.data.gouv.fr/opendata/datagouv/moderation/digest_daily/2021-09-18/2021-09-18.html).

**Comment by abulte** [comment]
Yes mais la grosse masse semble être en KM. Le top ce serait une requête prénom et nom finissent par deux majuscules mais je l'ai pas sous la main 😅

**Comment by maudetes** [comment]
Approximativement:
```
pattern = re.compile('[a-z][A-Z]{2}$')
us = User.objects.filter(first_name=pattern, last_name=pattern)
```

**Comment by abulte** [comment]
Autres constatations :
- en utilisant la requête ci-dessous, on remonte à peu près à juillet dernier (refonte graphique)
- quelques comptes (une dizaine) sont confirmés, c'est peut-être le but de la manoeuvre mais les comptes confirmés n'ont pas l'air utilisés
- PI le token de confirmation expire au bout de 5 jours https://flask-security-too.readthedocs.io/en/stable/configuration.html?highlight=confirmable#SECURITY_CONFIRM_EMAIL_WITHIN
- pas de `MembershipRequest` associée

**Comment by maudetes** [comment]
Trouver une remédiation à appliquer

**Comment by maudetes** [comment]
Priorisation de https://github.com/etalab/data.gouv.fr/issues/822 pour limiter la création d'utilisateurs comme vecteur de spam.
